### DISCUSSION TITLE
InstantStyle - The new IPAdapter reference implementation, style and composition are referenced separately

### DISCUSSION THREAD

**Discussion by xhoxye** [lllyasviel/Fooocus#2709] [Ideas]
https://github.com/InstantStyle/InstantStyle
InstantStyle: Free Lunch towards Style-Preserving in Text-to-Image Generation

https://github.com/cubiq/ComfyUI_IPAdapter_plus
https://www.youtube.com/watch?v=czcgJnoDVd4
![QQ截图20240405135543](https://github.com/lllyasviel/Fooocus/assets/129571231/21a19fe1-15a0-425a-8580-460248d7ce9c)
![QQ截图20240405135601](https://github.com/lllyasviel/Fooocus/assets/129571231/f0a2fe56-bcff-4e74-9dd8-823dddeaa074)
![QQ截图20240405135623](https://github.com/lllyasviel/Fooocus/assets/129571231/e64bf8d9-eb80-4a55-8439-ea73a8dee2be)

**Comment by haofanwang** [comment]
Thanks, looking forward it implementation in Fooocus.

**Comment by xhoxye** [comment]
https://github.com/Mikubill/sd-webui-controlnet/discussions/2770
ControlNet in A1111 is implemented today
